scapy/contrib/lldp.py

```diff
@@ -123,8 +124,13 @@ class LLDPDU(Packet):
 
     def guess_payload_class(self, payload):
         # type is a 7-bit bitfield spanning bits 1..7 -> div 2
-        lldpdu_tlv_type = orb(payload[0]) // 2
-        return LLDPDU_CLASS_TYPES[lldpdu_tlv_type]
```

Comment:
Use this instead:
`return LLDPDU_CLASS_TYPES.get(lldpdu_tlv_type, Raw)`

Reply:
addressed in c2aea83

Reply:
May I suggest to use `conf.raw_layer` instead?

Reply:
:+1: , see d7017d9

Reply:
while reading more code that forms the basement of scapy I came across ```default_payload_class()``` in ```class packet``` . should/could this be used here instead of
```conf.raw_layer``` or would this generate to much indirection, abstraction, ... ?